src/Illuminate/Database/Query/Builder.php

```diff
@@ -371,8 +371,8 @@ protected function prependDatabaseNameIfCrossDatabaseQuery($query)
             $this->getConnection()->getDatabaseName()) {
             $databaseName = $query->getConnection()->getDatabaseName();
 
-            if (strpos($query->from, $databaseName) !== 0 && strpos($query->from, '.') === false) {
-                $query->from($databaseName.'.'.$query->from);
+            if (strpos($query->getQuery()->from, $databaseName) !== 0 && strpos($query->getQuery()->from, '.') === false) {
+                $query->getQuery()->from($databaseName.'.'.$query->getQuery()->from);
```

Comment:
I'd suggest a small nitpick to cache `$query->getQuery()->from` to a variable since it's being used 3 times